### PR TITLE
SNOW-1324573: local testing fix flaky oob telemetry test

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1324573

the local testing oob telemetry is flaky because the service takes longer to process or throttled.
do the following to improve:
- increate the default wait time, this barely add overhead to customer as we are doing batch sending underneath
- add retry in our test
- remove duplicate and unreliable exact record tuple assertion, checking caplog text should be enough

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-jrose — APPROVED** [review]

**Comment by sfc-gh-aalam** [comment]
Is the test supposed to work without requests? `Exception response: None, exception: No module named 'requests'`

File: src/snowflake/snowpark/mock/_telemetry.py [hunk 1]
```diff
@@ -83,9 +87,8 @@ def _upload_payload(self, payload) -> None:
         success = True
         response = None
         try:
-            # import here is because stored proc doesn't have vendored request module
-            # have it at the top will cause import error in stored procedure running
-            from snowflake.connector.vendored import requests
+            # import here is because stored proc doesn't have requests as requirement
+            import requests
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/mock/_telemetry.py:91]: Question: Why do we use non vendored requests here?

File: tests/mock_unit/test_oob_telemetry.py [hunk 2]
```diff
@@ -139,28 +146,38 @@ def test_unit_oob_log_not_implemented_error(caplog, local_testing_telemetry_setu
 
     # test sending successfully
     caplog.clear()
-    with caplog.at_level(logging.DEBUG, logger="snowflake.snowpark.mock._telemetry"):
-        oob_service.log_not_supported_error(
-            external_feature_name=unraise_feature_name,
-        )
+
+    max_retry = 10
```
Comment (sfc-gh-stan) [inline tests/mock_unit/test_oob_telemetry.py:150]: What about replacing this with a pytest plugin: https://github.com/box/flaky?

File: src/snowflake/snowpark/mock/_telemetry.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/mock/_telemetry.py]: the vendored requests triggers unnecessary ocsp validation

File: tests/mock_unit/test_oob_telemetry.py (hunk 2 above)
Comment (sfc-gh-aalam) [reply tests/mock_unit/test_oob_telemetry.py:150]: this is very cool ^

**Review by sfc-gh-aalam — APPROVED** [review]

File: tests/mock_unit/test_oob_telemetry.py (hunk 2 above)
Comment (sfc-gh-aling) [reply tests/mock_unit/test_oob_telemetry.py:150]: this is good idea, but I did this retry manually for 2 reasons here:

1. I'm not sure introducing a new dependency (though just for test) would require any stored proc test change or other configure change. luckily nothing breaks and works, but we need extra time to test.
2. though it's test dependency, does it still need 3rd party library review process?

so I think we can log a JIRA for improvement at a later time.

**Comment by sfc-gh-aling** [comment]
@sfc-gh-aalam I'm not sure why doing a local import requests would trigger error ModuleNotFound.. but moving the import to global fixes the flakiness.

in stored proc requests is not available by default, I re-implement this so that stored proc shall also work.

**Pull request merged**